stdlib/3/io.pyi

```diff
@@ -20,6 +20,10 @@ _T = TypeVar('_T', bound=IOBase)
 
 open = builtins.open
 
+if sys.version_info >= (3, 8):
+    from os import PathLike
+    def open_code(path: Union[str, bytes, int, PathLike[Any]]) -> IO[bytes]: ...
```

Comment:
`int` doesn't seem to work:
```
In [3]: io.open_code(path=42)                                                                                                                                                      
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-576f3ecc9c97> in <module>
----> 1 io.open_code(path=42)

TypeError: open_code() argument 'path' must be str, not int
```

In fact from the C code it looks like it only accepts str:
```
_io.open_code

    path : unicode
```

Reply:
Yup, you're right. I got got by
> it should always be considered interchangeable with open(path, 'rb')

https://docs.python.org/3/library/io.html#io.open_code

Reply:
Kind of weird that they implemented it this way; it should clearly accept PathLike and probably should accept bytes paths too.

Reply:
Fourth BPO of the day! https://bugs.python.org/issue39723